h11/_receivebuffer.py

```diff
@@ -83,30 +91,56 @@ def maybe_extract_until_next(self, needle):
         # Returns extracted bytes on success (advancing offset), or None on
         # failure
         if self._looked_for == needle:
-            search_start = max(self._start, self._looked_at - len(needle) + 1)
+            looked_at = max(self._start, self._looked_at - len(needle) + 1)
         else:
-            search_start = self._start
-        offset = self._data.find(needle, search_start)
-        if offset == -1:
-            self._looked_at = len(self._data)
+            looked_at = self._start
             self._looked_for = needle
+            # re.compile operation is more expensive than just byte compare
```

Comment:
If I understand this comment correctly, you are justifying the check rather than just always compiling? If so maybe say `Check if default delimiter to avoid expensive re.compile`? As I was originally confused by this comment.

Reply:
Thank you, your explanation looks more clear (fixing)